kola/tests/sysext/disable_containerd.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2023 The Flatcar Maintainers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sysext
+
+import (
+	"github.com/coreos/go-semver/semver"
+	"github.com/flatcar/mantle/kola/cluster"
+	"github.com/flatcar/mantle/kola/register"
+	"github.com/flatcar/mantle/platform/conf"
+)
+
+func init() {
+	register.Register(&register.Test{
+		Run:         containerdDisable,
+		ClusterSize: 1,
+		Platforms:   []string{"qemu", "qemu-unpriv"},
+		Name:        "sysext.disable-containerd",
+		// Only releases after 3745 ship sysext
+		MinVersion: semver.Version{Major: 3746},
+		UserData: conf.Butane(`
+variant: flatcar
+version: 1.0.0
+storage:
+  links:
+  - path: /etc/extensions/containerd-flatcar.raw
+    target: /dev/null
+    hard: false
+    overwrite: true
+`),
+		Distros: []string{"cl"},
+	})
+}
+
+func containerdDisable(c cluster.TestCluster) {
+	m := c.Machines()[0]
+	output := c.MustSSH(m,
+		`test ! -f /usr/bin/containerd || echo "ERROR"`)
+	if string(output) == "ERROR" {
+		c.Errorf("/usr/bin/containerd exists even when sysext is disabled!")
+	}
```

Comment:
The test would be more robust by asserting what we expect instead of what we don't expect.
I think the easiest way would be:
```suggestion
	# Assert that the containerd binary does not exist
	_ = c.MustSSH(m, `test ! -f /usr/bin/containerd`)
```

Reply:
I disagree - the suggestion makes the command fail instead of creating a meaningful test outcome. This would imply the test itself is broken.

In general, tests should end in one of 3 ways:
1. The test succeeded, signalling the build is healthy (containerd is not present).
2. The test failed, signalling an issue with the build (containerd is present).
3. The test is broken, signalling an issue with the test suite (either "test" or "echo" are broken, or SSH failed.)

I want to communicate 2., not 3.

Reply:
Yes, it makes sense to distinguish that. The test is still not robust because it checks for errors explicitly. What you can change is to use `&& echo SUCCESS` and `if string(output) != "SUCCESS" {` to handle more things that can get wrong.

Reply:
I'll reverse the check and catch `test` returning 1 (if the binary is not present).

Reply:
Done, please have another look.

Reply:
What I meant was that `== "ERROR"` is too narrow and the test will silently break if it returns `ERROR\n` or the empty string. Therefore it's better to use something like `!= "SUCCESS"`.